rpc/item.c

```diff
@@ -548,6 +591,20 @@ M0_INTERNAL void m0_rpc_item_xid_assign(struct m0_rpc_item *item)
 	M0_LEAVE();
 }
 
+static void rpc_item_xid_unassign(struct m0_rpc_item *item)
```

Comment:
This function is called in m0_rpc_item_fini(), and we require the rpc machine lock is held.
But at the time of m0_rpc_item_fini() is invoked, must the rpc machine lock be held?

Or we can call this rpc_item_xid_unassign() when an item is replied?
e.g. in m0_rpc_item_process_reply()?

Reply:
> But at the time of m0_rpc_item_fini() is invoked, must the rpc machine lock be held?

If it's not held you will see the assertion failure down below, but it's not showing up.

AFAIR Initial version did unassign somewhere in `m0_rpc_item_process_reply()` or something similar. The problem is that the function may be called multiple times in case if reply is lost. We have such state transition in at least one UT.